TST random seed global /svm/tests/test_svm.py

Veghit opened the pull request:
Reference Issues/PRs
Towards https://github.com/scikit-learn/scikit-learn/issues/22827

What does this implement/fix? Explain your changes.
using random global seed for sklearn/svm/tests/test_svm.py to enable better testing.

Any other comments?
tests pass without logic changes. Decided to use global seed also for the test_svc_nonfinite_params test as it does not take much time and has some potential issues even though it is a simple test.

Commit:
typo fix

Commit:
more typo fixes

Commit:
more typo fixes

thomasjpfan committed:
DOC Allows section links to be visible when linked (#25783)

lesteve committed:
MNT Fix Pyodide build errors due to incompatible function pointer types (#25831)

genvalen committed:
MAINT parameter validation for sklearn.datasets.load_svmlight_files (#25811)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

lesteve committed:
CI Disable sphinx parallelism in CircleCI doc build (#25832)

Commit:
MAINT Parameters validation for metrics.recall_score (#25816)

tspeng committed:
MAINT Parameters validation for sklearn.datasets.fetch_olivetti_faces (#25823)

sortofamudkip committed:
MAINT Parameters validation for metrics.top_k_accuracy_score (#25828)

Co-authored-by: wishyut.pitawanik <wpitawanik@mpiwg-berlin.mpg.de>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

2357juan committed:
MAINT Use memoryviews in _random.pyx (#25780)

Co-authored-by: Juan Gomez <{789543}+{2357juan}@users.noreply.github.com>

adrinjalali committed:
MNT make plot_partial_dependence faster (#25806)

jeremiedbb committed:
MAINT Clean deprecated losses in (hist) gradient boosting for 1.3 (#25834)

jeremiedbb committed:
MAINT Clean deprecation of normalize in calibration_curve for 1.3 (#25833)

jeremiedbb committed:
BLD Clean command removes generated from cython templates (#25839)

Micky774 committed:
PERF Implement `PairwiseDistancesReduction` backend for `KNeighbors.predict_proba` (#24076)

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

ZeeshanLone committed:
MAINT Added Parameter Validation for datasets.make_circles (#25848)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

adrinjalali committed:
MNT use a single job by default with sphinx build (#25836)

jeremiedbb committed:
BLD Generate warning automatically for templated cython files (#25842)

Shivachauhan17 committed:
MAINT parameter validation for sklearn.datasets.fetch_lfw_people (#25820)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

AymericBasset committed:
MAINT Parameters validation for metrics.fbeta_score (#25841)

marenwestermann committed:
TST add global_random_seed fixture to sklearn/covariance/tests/test_robust_covariance.py (#25821)

choudharynishu committed:
MAINT Parameter validation for linear_model.orthogonal_mp (#25817)

glemaitre committed:
TST activate common tests for TSNE (#25374)

lesteve committed:
CI Update lock files (#25849)

ZeeshanLone committed:
MAINT Added Parameter Validation for metrics.mean_gamma_deviance (#25853)

hbenedek committed:
MAINT Parameters validation for feature_selection.mutual_info_regression (#25850)

pooja-subramaniam committed:
MAINT parameter validation metrics.class_likelihood_ratios (#25863)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jeremiedbb committed:
MAINT Ensure disjoint interval constraints (#25797)

pooja-subramaniam committed:
MAINT Parameters validation for utils.gen_batches (#25864)

rusdes committed:
TST use global_random_seed in test_dict_vectorizer.py (#24533)

yuanx749 committed:
TST use global_random_seed in test_pls.py (#24526)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

OmarManzoor committed:
TST use global_random_seed in test_gpc.py (#24600)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

2357juan committed:
DOC Fix overlapping plot axis in bench_sample_without_replacement.py (#25870)

2357juan committed:
MAINT Use contiguous memoryviews in _random.pyx (#25871)

Shivachauhan17 committed:
MAINT parameter validation sklearn.datasets.fetch_lfw_pair (#25857)

Commit:
MAINT Parameters validation for metrics.classification_report (#25868)

ogrisel committed:
Empty commit

glemaitre committed:
DOC fix docstring dtype parameter in OrdinalEncoder (#25877)

jeremiedbb committed:
MAINT Clean up depreacted "log" loss of SGDClassifier for 1.3 (#25865)

thomasjpfan committed:
ENH Adds TargetEncoder (#25334)

Co-authored-by: Andreas Mueller <t3kcit@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Jovan Stojanovic <62058944+jovan-stojanovic@users.noreply.github.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

ogrisel committed:
CI make it possible to cancel running Azure jobs (#25876)

jeremiedbb committed:
MAINT Clean-up deprecated if_delegate_has_method for 1.3 (#25879)

choudharynishu committed:
MAINT Parameter validation for tree.export_text (#25867)

rprkh committed:
DOC impact of `tol` for solvers in RidgeClassifier (#25530)

bharatr21 committed:
MAINT Parameters validation for metrics.hinge_loss (#25880)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

sortofamudkip committed:
MAINT Parameters validation for metrics.ndcg_score (#25885)

Commit:
using global random seeds instead of fixed seeds.

Commit:
using global random seeds instead of fixed seeds.

Commit:
black formatting

Commit:
black formatting

glevv committed:
ENH KMeans initialization account for sample weights (#25752)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Commit:
global random seed, replaced two functions with static seeds

Commit:
global random seed, replaced two functions with static seeds

Commit:
global random seed, replaced two functions with static seeds

Commit:
global random seed, replaced two functions with static seeds

Commit:
revert other changes

Commit:
revert other changes

Commit:
revert other changes

Commit:
revert other changes

Veghit committed:
Merge branch 'main' into TST_random_seed_global_svm

sortofamudkip committed:
TST use global_random_seed in sklearn/tests/test_dummy.py (#25884)

lorentzenchr committed:
DOC improve calibration user guide (#25687)

awinml committed:
ENH Support for sparse matrices added to `sklearn.metrics.silhouette_samples` (#24677)

Co-authored-by: Sahil Gupta <sahil@Sahils-MBP.lan>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

Veghit committed:
MAINT validate_params for plot_tree (#25882)

Co-authored-by: Itay <itayvegh@gmail.com>

precondition committed:
MAINT add missing space in error message in SVM (#25913)

thomasjpfan committed:
FIX Adds requires_y tag to TargetEncoder (#25917)

jeremiedbb committed:
MAINT Consistent cython types continued (#25810)

jeremiedbb committed:
TST Speed-up common tests of DictionaryLearning (#25892)

jeremiedbb committed:
TST Speed-up test_dbscan_optics_parity (#25893)

marctorsoc committed:
ENH add np.nan option for zero_division in precision/recall/f-score (#25531)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

Commit:
MAINT Parameters validation for datasets.make_low_rank_matrix (#25901)

ZeeshanLone committed:
MAINT Parameter validation for metrics.cluster.adjusted_mutual_info_score (#25898)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jeremiedbb committed:
TST Speed-up test_partial_dependence.test_output_shape (#25895)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

Commit:
MAINT Parameters validation for datasets.make_regression (#25899)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jeremiedbb commented on sklearn/svm/tests/test_svm.py:
```diff
@@ -1070,11 +1070,11 @@ def test_linear_svm_convergence_warnings():
     # Non-regression test for #2933.
-    X = np.random.RandomState(21).randn(10, 3)
-    y = np.random.RandomState(12).randn(10)
+    X = np.random.RandomState(global_random_seed).randn(10, 3)
+    y = np.random.RandomState(global_random_seed + 1).randn(10)
```
```suggestion
    rng = np.random.RandomState(global_random_seed)
    X = rng.randn(10, 3)
    y = rng.randn(10)
```

jeremiedbb reviewed (commented):
Thanks for the PR @Veghit. There are many tests in this file that could use the global_random_seed since they currently use a fixed ``random_state``

DominicFox committed:
MAINT Parameters validation for metrics.mean_squared_log_error (#25924)

Veghit committed:
TST Use global_random_seed in tests/test_naive_bayes.py (#25890)

marenwestermann committed:
TST add global_random_seed fixture to sklearn/datasets/tests/test_covtype.py (#25904)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

Commit:
MAINT Parameters validation for datasets.make_multilabel_classification (#25920)

Commit:
adding global random seeds instead of static randoms

Commit:
adding global random seeds instead of static randoms

Commit:
adding global random seeds instead of static randoms

Commit:
adding global random seeds instead of static randoms

futurewarning committed:
Fixed feature mapping typo (#25934)

glemaitre committed:
MAINT switch to newer codecov uploader (#25919)

Co-authored-by: Loïc Estève <loic.esteve@ymail.com>

jeremiedbb committed:
TST Speed-up test suite when using pytest-xdist (#25918)

adrinjalali committed:
DOC update license year to 2023 (#25936)

Charlie-XIAO committed:
FIX Remove spurious feature names warning in IsolationForest (#25931)

ogrisel committed:
TST fix unstable test_newrand_set_seed (#25940)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jeremiedbb committed:
MAINT Clean-up deprecated max_features="auto" in trees/forests/gb (#25941)

joseortiz3 committed:
MAINT LogisticRegression informative error msg when penaly=elasticnet and l1_ratio is None (#25925)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

jeremiedbb committed:
MAINT Clean-up remaining SGDClassifier(loss="log") (#25938)

thomasjpfan committed:
FIX Fixes pandas extension arrays in check_array (#25813)

thomasjpfan committed:
FIX Fixes pandas extension arrays with objects in check_array (#25814)

jeremiedbb committed:
CI Disable pytest-xdist in pylatest_pip_openblas_pandas build (#25943)

glemaitre committed:
MAINT remove deprecated call to resources.content (#25951)

lorentzenchr committed:
DOC note on calibration impact on ranking (#25900)

betatim committed:
Remove loguniform fix, use scipy.stats instead (#24665)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

ChVeen committed:
MAINT Fix broken links in cluster.dbscan module (#25958)

adienes committed:
DOC Fix lars Xy shape (#25952)

dberenbaum committed:
ENH Add drop_intermediate parameter to metrics.precision_recall_curve (#24668)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lene committed:
FIX improve error message when computing NDCG with a single document (#25672)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

glemaitre committed:
MAINT introduce _get_response_values and _check_response_methods (#23073)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jjerphan committed:
MAINT Extend message for large sparse matrices support (#25961)

Co-authored-by: Meekail Zain <34613774+Micky774@users.noreply.github.com>

Commit:
MAINT Parameters validation for datasets.make_gaussian_quantiles (#25959)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

A-H-Mansoury committed:
MAINT Parameters validation for sklearn.metrics.d2_tweedie_score (#25975)

Commit:
MAINT Parameters validation for datasets.make_hastie_10_2 (#25967)

yuanx749 committed:
MAINT Parameters validation for preprocessing.minmax_scale (#25962)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Commit:
MAINT Parameters validation for datasets.make_checkerboard (#25955)

Commit:
MAINT Parameters validation for datasets.make_biclusters (#25945)

Commit:
MAINT Parameters validation for datasets.make_moons (#25971)

lorentzenchr committed:
DOC replace deviance by loss in docstring of GradientBoosting (#25968)

ChVeen committed:
MAINT Fix broken link in feature_selection/_univariate_selection.py (#25984)

Lothiraldan committed:
DOC Update model_persistence.rst to fix skops example (#25993)

Co-authored-by: adrinjalali <adrin.jalali@gmail.com>

pinkgoldpeach committed:
DOC Specified meaning for max_patches=None in extract_patches_2d  (#25996)

windiana42 committed:
DOC document that last step is never cached in pipeline (#25995)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

Charlie-XIAO committed:
FIX SequentialFeatureSelector throws IndexError when cv is a generator (#25973)

thomasjpfan committed:
ENH Adds infrequent categories support to OrdinalEncoder (#25677)

Co-authored-by: Tim Head <betatim@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Andreas Mueller <t3kcit@gmail.com>

mdarii committed:
MAINT make plot_digits_denoising deterministic by fixing random state (#26004)

murezzda committed:
DOC improve example of PatchExtractor (#26002)

Commit:
MAINT Parameters validation for datasets.make_friedman2 (#25986)

Commit:
MAINT Parameters validation for datasets.make_friedman3 (#25989)

Commit:
MAINT Parameters validation for datasets.make_sparse_uncorrelated (#26001)

Commit:
MAINT Parameters validation for datasets.make_spd_matrix (#26003)

Commit:
MAINT Parameters validation for datasets.make_sparse_spd_matrix (#26009)

ppiont committed:
DOC Added the meanings of default=None for PatchExtractor parameters (#26005)

glemaitre committed:
MAINT remove unecessary check covered by parameter validation framework (#26014)

jeremiedbb committed:
MAINT Consistent cython types from _typedefs (#25942)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

Commit:
MAINT Parameters validation for datasets.make_swiss_roll (#26020)

Commit:
MAINT Parameters validation for datasets.make_s_curve (#26022)

Commit:
MAINT Parameters validation for datasets.make_blobs (#25983)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

Charlie-XIAO committed:
DOC fix SplineTransformer include_bias docstring (#26018)

Charlie-XIAO committed:
ENH RocCurveDisplay add option to plot chance level (#25987)

glemaitre committed:
DOC show from_estimator and from_predictions for Displays (#25994)

lesteve committed:
EXA Fix rst in plot_partial_dependence (#26028)

thomasjpfan committed:
CI Adds coverage to docker jobs on Azure (#26027)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

jpangas committed:
API Replace `n_iter` in `Bayesian Ridge` and `ARDRegression` (#25697)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

thomasjpfan committed:
CLN Make _NumPyAPIWrapper naming consistent to _ArrayAPIWrapper (#26039)

ogrisel committed:
CI disable coverage on Windows to keep CI times reasonable (#26052)

lesteve committed:
DOC Use Scientific Python Plausible instance for analytics (#25547)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.preprocessing.scale (#26036)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.haversine_distances (#26047)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.laplacian_kernel (#26048)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.linear_kernel (#26049)

dmitrylala committed:
MAINT Parameters validation for sklearn.metrics.silhouette_samples (#26053)

ROMEEZHOU committed:
MAINT Parameters validation for sklearn.preprocessing.add_dummy_feature (#26058)

ZeeshanLone committed:
Added Parameter Validation for metrics.cluster.normalized_mutual_info_score() (#26060)

ChVeen committed:
DOC Typos in HistGradientBoosting documentation (#26057)

marenwestermann committed:
TST add global_random_seed fixture to sklearn/datasets/tests/test_rcv1.py (#26043)

ekaterinabutyugina committed:
MAINT Parameters validation for sklearn.metrics.pairwise.cosine_similarity (#26006)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

thomasjpfan committed:
ENH Adds isdtype to Array API wrapper (#26029)

dmitrylala committed:
MAINT Parameters validation for sklearn.metrics.silhouette_score (#26054)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

glemaitre committed:
FIX fix spelling mistake in _NumPyAPIWrapper (#26064)

ogrisel committed:
CI ignore more non-library Python files in codecov (#26059)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.cosine_distances (#26046)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

glemaitre committed:
MAINT Introduce BinaryClassifierCurveDisplayMixin (#25969)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

thomasjpfan committed:
ENH Forces shape to be tuple when using Array API's reshape (#26030)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Tim Head <betatim@gmail.com>

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.paired_euclidean_distances (#26073)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.paired_manhattan_distances (#26074)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.paired_cosine_distances (#26075)

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.preprocessing.binarize (#26076)

sortofamudkip committed:
MAINT Parameters validation for metrics.explained_variance_score (#26079)

glemaitre committed:
DOC use correct template name for displays (#26081)

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.preprocessing.maxabs_scale (#26077)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.preprocessing.label_binarize (#26078)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

ansamz committed:
MAINT parameter validation for d2_absolute_error_score (#26066)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

ansamz committed:
MAINT Parameter validation for roc_auc_score (#26007)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.preprocessing.normalize (#26069)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

ZeeshanLone committed:
MAINT Parameter validation for metrics.cluster.fowlkes_mallows_score (#26080)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

genvalen committed:
MAINT Parameters validation for compose.make_column_transformer (#25897)

Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.polynomial_kernel (#26070)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.rbf_kernel (#26071)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.metrics.pairwise.sigmoid_kernel (#26072)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jeremiedbb committed:
MAINT Param validation: constraint for numeric missing values (#26085)

thomasjpfan committed:
FIX Adds support for negative values in categorical features in gradient boosting (#25629)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>
Co-authored-by: Tim Head <betatim@gmail.com>

ChVeen committed:
MAINT Fix C warning in Cython module splitting.pyx (#26051)

BabaYaga1221 committed:
MNT Updates _isotonic.pyx to use memoryviews instead of `cnp.ndarray` (#26068)

thomasjpfan committed:
FIX Fixes memory regression for inspecting extension arrays (#26106)

ogrisel committed:
PERF set openmp to use only physical cores by default (#26082)

adrinjalali committed:
MNT Update black to 23.3.0 (#26110)

adrinjalali committed:
MNT Adds black commit to git-blame-ignore-revs (#26111)

ROMEEZHOU committed:
MAINT Parameters validation for sklearn.metrics.pair_confusion_matrix (#26107)

ROMEEZHOU committed:
MAINT Parameters validation for sklearn.metrics.mean_poisson_deviance (#26104)

ArturoAmorQ committed:
DOC Use notebook style in plot_lof_outlier_detection.py (#26017)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jeremiedbb committed:
MAINT utils._fast_dict uses types from utils._typedefs (#26025)

rprkh committed:
DOC remove sparse-matrix for `y` in ElasticNet (#26127)

lorentzenchr committed:
ENH add exponential loss (#25965)

2357juan committed:
MAINT Parameters validation for sklearn.preprocessing.robust_scale (#26086)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.datasets.fetch_rcv1 (#26126)

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.metrics.adjusted_rand_score (#26134)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.metrics.calinski_harabasz_score  (#26135)

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.metrics.davies_bouldin_score  (#26136)

rgommers committed:
MAINT: remove `from numpy.math cimport` statements (#26143)

yuanx749 committed:
MAINT Parameters validation for sklearn.inspection.permutation_importance (#26145)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Jessicakk0711 committed:
MAINT Parameters validation for sklearn.metrics.cluster.homogeneity_completeness_v_measure (#26137)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

jiawei-zhang-a committed:
MAINT Parameters validation for sklearn.metrics.rand_score (#26138)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

SinghAnkur28 committed:
DOC update comment in metrics/tests/test_classification.py (#26150)

ogrisel committed:
CI small cleanup of Cirrus CI test script (#26168)

glemaitre committed:
MAINT remove deprecated is_categorical_dtype (#26156)

seoeunHong committed:
DOC Add skforecast to related projects page (#26133)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
FIX Keeps namedtuple's class when transform returns a tuple (#26121)

SinghAnkur28 committed:
DOC corrected letter case for better readability in sklearn/metrics/_classification.py / (#26169)

xi-jiajun committed:
MAINT Parameters validation for sklearn.preprocessing.power_transform (#26142)

rprkh committed:
FIX `roc_auc_score` now uses `y_prob` instead of `y_pred` (#26155)

dvignesh1995 committed:
MAINT Parameters validation for sklearn.datasets.load_iris (#26177)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.datasets.load_diabetes (#26166)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.datasets.load_breast_cancer (#26165)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

Vincent-violet committed:
MAINT Parameters validation for sklearn.metrics.cluster.entropy (#26162)

Charlie-XIAO committed:
MAINT Parameters validation for sklearn.datasets.fetch_species_distributions (#26161)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

OmarManzoor committed:
ASV Fix tol in SGDRegressorBenchmark (#26146)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>

lesteve committed:
MNT use api.openml.org URLs for fetch_openml (#26171)

elabongaatuo committed:
MAINT Parameters validation for sklearn.utils.resample (#26139)

glemaitre committed:
MAINT make it explicit that additive_chi2_kernel does not accept sparse matrix (#26178)

adrinjalali committed:
MNT fix circleci link in README.rst (#26183)

jeremiedbb committed:
CI Fix circleci artifact redirector action (#26181)

Commit:
multiple test fixes to accommodate the new global_random_seed with lower thresholds and other fixes.

Commit:
multiple test fixes to accommodate the new global_random_seed with lower thresholds and other fixes.

Commit:
multiple test fixes to accommodate the new global_random_seed with lower thresholds and other fixes.

Commit:
multiple test fixes to accommodate the new global_random_seed with lower thresholds and other fixes.

Commit:
using a single RandomState instead of 2 where possible

Commit:
using a single RandomState instead of 2 where possible

Commit:
using a single RandomState instead of 2 where possible

Commit:
fixed 2 issues after pipeline failed

Commit:
fixed 2 issues after pipeline failed

Commit:
fixed 2 issues after pipeline failed

Commit:
fixed 2 issues after pipeline failed

Commit:
Merge remote-tracking branch 'origin/TST_random_seed_global_svm' into TST_random_seed_global_svm

# Conflicts:
#	sklearn/svm/tests/test_svm.py

Commit:
black plugin

Commit:
black plugin

Commit:
revert NB

Commit:
revert NB

Commit:
Merge remote-tracking branch 'origin/TST_random_seed_global_svm' into TST_random_seed_global_svm

# Conflicts:
#	sklearn/svm/tests/test_svm.py

Commit:
line too long fix

Veghit commented:
@jeremiedbb fixed the rest of the tests

jeremiedbb committed:
Merge remote-tracking branch 'upstream/main' into pr/Veghit/25891

jeremiedbb committed:
cln [all random seeds]
test_libsvm_iris
test_precomputed
test_linearsvr_fit_sampleweight
test_oneclass_decision_function
test_probability
test_decision_function
test_decision_function_shape
test_svr_predict
test_weight
test_auto_weight
test_bad_input
test_svc_nonfinite_params
test_unicode_kernel
test_linearsvc_parameters
test_linearsvc
test_linearsvc_crammer_singer
test_linearsvc_fit_sampleweight
test_crammer_singer_binary
test_linearsvc_iris
test_liblinear_set_coef
test_immutable_coef_property
test_svc_clone_with_callable_kernel
test_libsvm_convergence_warnings
test_consistent_proba
test_linear_svm_convergence_warnings
test_svr_coef_sign
test_hasattr_predict_proba
test_decision_function_shape_two_class
test_svc_invalid_break_ties_param
test_svc_ovr_tie_breaking
test_linearsvm_liblinear_sample_weight
test_n_iter_libsvm
test_svm_with_infinite_C

jeremiedbb committed:
iter [all random seeds]
    test_libsvm_iris
    test_precomputed
    test_linearsvr_fit_sampleweight
    test_oneclass_decision_function
    test_probability
    test_decision_function
    test_decision_function_shape
    test_svr_predict
    test_weight
    test_auto_weight
    test_bad_input
    test_svc_nonfinite_params
    test_unicode_kernel
    test_linearsvc_parameters
    test_linearsvc
    test_linearsvc_crammer_singer
    test_linearsvc_fit_sampleweight
    test_crammer_singer_binary
    test_linearsvc_iris
    test_liblinear_set_coef
    test_immutable_coef_property
    test_svc_clone_with_callable_kernel
    test_libsvm_convergence_warnings
    test_consistent_proba
    test_linear_svm_convergence_warnings
    test_svr_coef_sign
    test_hasattr_predict_proba
    test_decision_function_shape_two_class
    test_svc_invalid_break_ties_param
    test_svc_ovr_tie_breaking
    test_linearsvm_liblinear_sample_weight
    test_n_iter_libsvm
    test_svm_with_infinite_C

jeremiedbb committed:
iter [all random seeds]
    test_libsvm_iris
    test_precomputed
    test_linearsvr_fit_sampleweight
    test_oneclass_decision_function
    test_probability
    test_decision_function
    test_decision_function_shape
    test_svr_predict
    test_weight
    test_auto_weight
    test_bad_input
    test_svc_nonfinite_params
    test_unicode_kernel
    test_linearsvc_parameters
    test_linearsvc
    test_linearsvc_crammer_singer
    test_linearsvc_fit_sampleweight
    test_crammer_singer_binary
    test_linearsvc_iris
    test_liblinear_set_coef
    test_immutable_coef_property
    test_svc_clone_with_callable_kernel
    test_libsvm_convergence_warnings
    test_consistent_proba
    test_linear_svm_convergence_warnings
    test_svr_coef_sign
    test_hasattr_predict_proba
    test_decision_function_shape_two_class
    test_svc_invalid_break_ties_param
    test_svc_ovr_tie_breaking
    test_linearsvm_liblinear_sample_weight
    test_n_iter_libsvm
    test_svm_with_infinite_C

jeremiedbb committed:
Merge branch 'main' into TST_random_seed_global_svm

jeremiedbb committed:
iter [all random seeds]
    test_libsvm_iris
    test_precomputed
    test_linearsvr_fit_sampleweight
    test_probability
    test_decision_function
    test_decision_function_shape
    test_svr_predict
    test_bad_input
    test_svc_nonfinite_params
    test_unicode_kernel
    test_linearsvc_parameters
    test_linearsvc
    test_linearsvc_crammer_singer
    test_linearsvc_fit_sampleweight
    test_crammer_singer_binary
    test_linearsvc_iris
    test_liblinear_set_coef
    test_immutable_coef_property
    test_svc_clone_with_callable_kernel
    test_libsvm_convergence_warnings
    test_consistent_proba
    test_linear_svm_convergence_warnings
    test_svr_coef_sign
    test_hasattr_predict_proba
    test_decision_function_shape_two_class
    test_svc_invalid_break_ties_param
    test_svc_ovr_tie_breaking
    test_linearsvm_liblinear_sample_weight
    test_n_iter_libsvm
    test_svm_with_infinite_C

jeremiedbb committed:
Merge remote-tracking branch 'Veghit/TST_random_seed_global_svm' into pr/Veghit/25891

jeremiedbb committed:
check [all random seeds]
    test_libsvm_iris
    test_precomputed
    test_linearsvr_fit_sampleweight
    test_probability
    test_decision_function
    test_decision_function_shape
    test_svr_predict
    test_bad_input
    test_svc_nonfinite_params
    test_unicode_kernel
    test_linearsvc_parameters
    test_linearsvc
    test_linearsvc_crammer_singer
    test_linearsvc_fit_sampleweight
    test_crammer_singer_binary
    test_linearsvc_iris
    test_liblinear_set_coef
    test_immutable_coef_property
    test_svc_clone_with_callable_kernel
    test_libsvm_convergence_warnings
    test_consistent_proba
    test_linear_svm_convergence_warnings
    test_svr_coef_sign
    test_hasattr_predict_proba
    test_decision_function_shape_two_class
    test_svc_invalid_break_ties_param
    test_svc_ovr_tie_breaking
    test_linearsvm_liblinear_sample_weight
    test_n_iter_libsvm
    test_svm_with_infinite_C

jeremiedbb committed:
fix

jeremiedbb committed:
fix

jeremiedbb reviewed (approved):
LGTM. Thanks !

jeremiedbb committed:
Merge branch 'main' into TST_random_seed_global_svm
